### DISCUSSION TITLE
I am planning to use a Pre-build Sparkle as a Binary for my app.

### DISCUSSION THREAD

**Discussion by nishaanth-ms-21155** [sparkle-project/Sparkle#2598] [Q&A] [answered]
Is Sparkle's binary universal, as in does a Binary built on x64 work on arm64? I tested it and it work and I don't know why. Does Sparkle generate a universal binary by default? Or do I need to generate separate binaries for any reason in a production environment?

**Comment by zorgiepoo** [comment, marked as answer]
We distribute prebuilt universal binaries of the framework and the project is configured to build for both architectures. Sparkle currently supports a deployment target back to 10.13 which was before Apple silicon. You’d have to go out of your way to configure the project to not build for x86_64. You can also run ‘file’ command on a binary to see what architectures it is built for.

**Reply by nishaanth-ms-21155** [reply]
I have used deployment target 10.13 yet it works for Apple silicon. It is fine that it works. But is there any reason I should use the latest binary since the app already is configured to use the an older version where sparkle is built each time?
Can you clarify how do I use the file command on the generated binary?

**Comment by zorgiepoo** [comment]
```
> file Sparkle.framework/Sparkle 
Sparkle.framework/Sparkle: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
Sparkle.framework/Sparkle (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
Sparkle.framework/Sparkle (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

Sparkle like most projects sets `ARCHS` to `ARCHS_STANDARD` which evaluates to `arm64 x86_64` these days (i.e a universal binary).

> But is there any reason I should use the latest binary since the app already is configured to use the an older version where sparkle is built each time?

I don't understand this question. Is it asking about using the latest version of Sparkle, or asking about bumping the deployment target of Sparkle from 10.13 to a higher version (in which case you cannot use a prebuilt distribution of Sparkle)?

We distribute prebuilt binaries throughout our package managers (SPM, CocoaPods, etc). Xcode doesn't support building binaries with a deployment target less than 10.13 these days.

**Reply by nishaanth-ms-21155** [reply]
Let me clarify. Does DEPLOYMENT target of 10.13 using which I am compiling sparkle v1.24.0 support arm64 if I had compiled it in x64?
Kindly help. Thanks.

**Reply by zorgiepoo** [reply]
It does not matter if the host machine you are compiling from is an Intel or Apple Silicon Mac. It only matters what architectures the project is configured to build for. It is `ARCHS_STANDARD` which today means `arm64` and `x86_64`, i.e, a universal binary. (Technical hidden detail: a deployment target of 10.13 actually sets the minimum OS of the x86_64 slice to 10.13, while setting the minimum OS of the arm64 slice to 11.0 (when arm64 support was first introduced)).

I also recommend updating Sparkle, 1.24.0 is pretty old and missing security fixes at this point.
